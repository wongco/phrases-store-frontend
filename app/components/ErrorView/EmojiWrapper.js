import styled from 'styled-components';

const EmojiWrapper = styled.div`
  font-size: 150%;
  margin: 0 0.5em;
`;

export default EmojiWrapper;
