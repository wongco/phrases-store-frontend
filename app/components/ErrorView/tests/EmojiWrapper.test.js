import React from 'react';
import { enzymeFind } from 'styled-components/test-utils';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import EmojiWrapper from '../EmojiWrapper';

describe('<EmojiWrapper />', () => {
  it('renders without crashing', () => {
    shallow(<EmojiWrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<EmojiWrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<EmojiWrapper>{children}</EmojiWrapper>);
    const renderedComponent = enzymeFind(wrapper, EmojiWrapper);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
