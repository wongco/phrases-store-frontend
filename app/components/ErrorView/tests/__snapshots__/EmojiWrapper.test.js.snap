// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiWrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "EmojiWrapper-sc-57wo0l-0",
        "isStatic": true,
        "rules": Array [
          "font-size:150%;margin:0 0.5em;",
        ],
      },
      "displayName": "EmojiWrapper",
      "render": [Function],
      "styledComponentId": "EmojiWrapper-sc-57wo0l-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
